### ISSUE TITLE
Value injector for RESTEasy

### ISSUE THREAD

**Issue by victornoel** [pac4j/jax-rs-pac4j#6]
For now value injection with `@Pac4JProfile` and `@Pac4JProfileManager` only works with Jersey.

**Comment by rkorn86** [comment]
Is there currently any workaround to use resteasy with `@Pac4JCallback`? Im stuck on Wildfly 10 and RESTeasy.

**Comment by victornoel** [comment]
@rkorn86 there is no problem with `@Pac4JCallbac` with resteasy… or at least it doesn't concern this issue!

Why don't you come explain your problem in details on the pac4j-user mailing list? It would be easier to help you :)

**Comment by rkorn86** [comment]
Done - https://groups.google.com/forum/#!topic/pac4j-users/-dv6JPe9v8Q

**Comment by victornoel** [comment]
To implement this, we would need to implement something like this: https://dzone.com/articles/annotations-your-way-resteasy

**Comment by victornoel** [comment]
Or maybe better: https://github.com/resteasy/Resteasy/blob/master/testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/custom/CustomValueInjectorTest.java

**Comment by victornoel** [comment]
@rkorn86 I will think of workaround tomorrow :)

**Comment by victornoel** [comment]
Ok, so here are some ideas to get the current profile inside a resource method:

If you are running on top of a servlet container, you can very simply exploits the `HttpServletRequest` as explained [there](https://github.com/pac4j/j2e-pac4j#5-get-the-user-profile-profilemanager):
```java
@GET
public void get(@Context HttpServletRequest request) {
    ProfileManager manager = new ProfileManager(new J2EContext(request, null));
    Optional<CommonProfile> profile = manager.get(true);
}
```

If not, then it should be possible but a bit more complex, I will think about it in the future…

**Comment by rkorn86** [comment]
Workaround works like a charm. Sorry, but I dont have the time/knowledge to help you currently.

**Comment by victornoel** [comment]
Great, I will add it to the README. No problem for the help :)

**Comment by victornoel** [comment]
Closed by #31 :) Thanks again @yegorius
